http/get_simple/java/server/src/main/java/com/example/ArrowHttpServer.java

```diff
@@ -0,0 +1,131 @@
+
+public class ArrowHttpServer extends AbstractHandler {
+
+    static BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+
+    static Schema schema = new Schema(
+        List.of(
+            new Field("a", FieldType.nullable(new ArrowType.Int(64, true)), null),
+            new Field("b", FieldType.nullable(new ArrowType.Int(64, true)), null),
+            new Field("c", FieldType.nullable(new ArrowType.Int(64, true)), null),
+            new Field("d", FieldType.nullable(new ArrowType.Int(64, true)), null)
+        ));
+
+    static List<ArrowRecordBatch> batches;
+
+    static Random random = new Random();
+
+    public static List<ArrowRecordBatch> getPutData() {
+        int totalRecords = 100000000;
+        int length = 4096;
+
+        List<ArrowRecordBatch> batches = new ArrayList<>();
+
+        try (VectorSchemaRoot root = VectorSchemaRoot.create(schema, allocator)) {
+            
+            String[] names = schema.getFields().stream().map(Field::getName).toArray(String[]::new);
+            for (String name : names) {
+                byte[] randomBytes = new byte[length * 8];
+                random.nextBytes(randomBytes);
+
+                byte[] validityBytes = new byte[length / 8];
+                Arrays.fill(validityBytes, (byte) 0xFF);
+
+                BigIntVector vector = (BigIntVector) root.getVector(name);
+                vector.allocateNew(length);
+                vector.setValueCount(length);
+                ArrowBuf dataBuffer = vector.getDataBuffer();
+                dataBuffer.setBytes(0, randomBytes);
+
+                ArrowBuf validityBuffer = vector.getValidityBuffer();
+                validityBuffer.setBytes(0, validityBytes);
+                root.setRowCount(length);
+            }
+
+            int records = 0;
+            int lastLength;
+            while (records < totalRecords) {
+                if (records + length > totalRecords) {
+                    lastLength = totalRecords - records;
+                    try (VectorSchemaRoot slice = root.slice(0, lastLength)) {
+                        VectorUnloader unloader = new VectorUnloader(slice);
+                        ArrowRecordBatch arb = unloader.getRecordBatch();
+                        batches.add(arb);
+                    }
+                    records += lastLength;
+                } else {
+                    VectorUnloader unloader = new VectorUnloader(root);
+                    ArrowRecordBatch arb = unloader.getRecordBatch();
+                    batches.add(arb);
+                    records += length;
+                }
+            }
+        }
+
+        return batches;
+    }
+
+    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+
+        response.setContentType("application/vnd.apache.arrow.stream");
+        response.setStatus(HttpServletResponse.SC_OK);
+
```

Comment:
I'd be interested to see if we can figure out the `Content-Length` beforehand.  Given this is meant to be an example, I would find this useful, as I believe it's best practice to include the `Content-Length` header whenever it's feasible to calculate it in order to avoid chunked responses.

Reply:
I think effectively in Arrow you have to actually write the data to get the final length (since you have to construct the metadata, pad buffers, etc.). For this example that's doable since it's fixed data.

Reply:
@tdcmeehan thanks — I'm building a list of follow-ups to investigate and I will add this to it

Reply:
@lidavidm thanks.  Pardon my ignorance, but would it be possible for Arrow to provide an accurate estimate of the final length in advance of writing it out?  For example, infer buffers needed based on the content, estimate the metadata size needed, etc?

Reply:
@tdcmeehan In the case where the data that you're writing out is fully materialized before you start writing it out, we could just create the IPC messages and count how many bytes are in them to provide an exact `Content-Length` before writing them. But in the case where the data is not fully materialized before you start writing it out, it would be extremely difficult to estimate the final content length. It might be easy enough to estimate it with toy examples, but in real world cases where you have dictionaries and other encodings, there's no simple approach to that and there's no tooling in any of the Arrow libraries AFAIK that does estimations like that.